src/main/java/organice/logic/commands/MatchCommand.java

```diff
@@ -0,0 +1,90 @@
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.Person;
+import organice.model.person.TissueType;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Lists all persons in the address book to the user.
+ */
+public class MatchCommand extends Command {
+
+    public static final String COMMAND_WORD = "match";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Matches patients and donors who passes matching tests"
+            + "Parameters: all/ (to match all patients and donors)"
+            + "Parameters: ic/NRIC (to match a patient of specified NRIC)";
+
+    public static final String MESSAGE_SUCCESS = "Matched all patients and donors";
+    public static final String MESSAGE_PERSON_NOT_FOUND = "The patient with Nric %1$s cannot be found in ORGANice!";
+
+    public static final Double SUCCESSFUL_PERCENTAGE = 80.0;
+
+
+    private String input;
+    private Patient patient;
+
+    public MatchCommand(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Returns true if the donor and patient is a match.
+     * A match happens when the donor and patient's blood type and tissue types are compatible.
+     * @param donor {@code Donor} who will tested for suitability of donation
+     * @param patient {@code Patient} in need of organ
+     */
+    public boolean match(Person donor, Patient patient) {
+        if (!(donor instanceof Donor)) {
+            return false;
+        }
+
+        BloodType donorBloodType = ((Donor) donor).getBloodType();
+        BloodType patientBloodType = patient.getBloodType();
+
+        TissueType donorTissueType = ((Donor) donor).getTissueType();
+        TissueType patientTissueType = patient.getTissueType();
+
+        if (patientBloodType.isBloodTypeMatch(donorBloodType)
+                && donorTissueType.getPercentageMatch(patientTissueType) >= SUCCESSFUL_PERCENTAGE) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+
+        Nric patientNric = new Nric(input);
+        try {
+            if (model.hasPatient(patientNric)) {
+                patient = model.getPatient(patientNric);
+                Predicate<Person> matchesWithNric = donor -> match(donor, patient);
+                model.updateFilteredPersonList(matchesWithNric);
+            }
+            return new CommandResult(MESSAGE_SUCCESS);
+        } catch (PersonNotFoundException pne) {
+            return new CommandResult(String.format(MESSAGE_PERSON_NOT_FOUND, patientNric));
+        }
+
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof MatchCommand // instanceof handles nulls
+                && input.equals(((MatchCommand) other).input)); // state check
```

Comment:
since you dont use `requireNonNull` for the MatchCommand constructor u need to handle `null`, can check `ListCommand` on how to handle the null

Reply:
Added `requireNonNull` in the constructor